### PR TITLE
feat: swap SELL and BUY fields

### PR DESCRIPTION
**Pull request by nenadV91**
# Summary

This PR adds functionality to switch token fields.
The `outputCurrencyAmount` will be moved to `inputCurrencyAmount` and the BUY amount will be recalculated with the new quote.

# To test
- go to advanced orders widget
- test the switch tokens button

### CODE REVIEW

File: src/modules/advancedOrders/hooks/useAdvancedOrdersActions.ts [hunk 1]
```diff
@@ -46,10 +57,31 @@ export function useAdvancedOrdersActions() {
     console.log('On change recipient')
   }, [])
 
-  // TODO: implement this one
   const onSwitchTokens = useCallback(() => {
-    console.log('On switch tokens')
-  }, [])
+    if (!isWrapOrUnwrap) {
```
Comment (shoom3301) [inline src/modules/advancedOrders/hooks/useAdvancedOrdersActions.ts:61]: The code is very similar with `useLimitOrdersWidgetActions`, can we unify it?

Comment (nenadV91) [reply src/modules/advancedOrders/hooks/useAdvancedOrdersActions.ts:61]: I would suggest we keep it separate.

Comment (shoom3301) [reply src/modules/advancedOrders/hooks/useAdvancedOrdersActions.ts:61]: Why?
It it violates the DRY principle

Comment (anxolin) [reply src/modules/advancedOrders/hooks/useAdvancedOrdersActions.ts:61]: @nenadV91 is good you defend your views, but it helps if you give some arguments. 

In principle, this seems a clear case of code duplication, and this leads to errors. It's very easy to fix in the future something for the limit orders logic, and forget to do the same in twap.

Comment (nenadV91) [reply src/modules/advancedOrders/hooks/useAdvancedOrdersActions.ts:61]: There are some differences between those two

For example in twap orders we have
```javascript
      updateQuoteState({ response: null })
      updateAdvancedOrderRawState({
        inputCurrencyId: outputCurrencyId,
        outputCurrencyId: inputCurrencyId,
        inputCurrencyAmount: FractionUtils.serializeFractionToJSON(outputCurrencyAmount),
        outputCurrencyAmount: null,
      })
```

And in limit orders
```javascript
      updateLimitOrdersState({
        inputCurrencyId: outputCurrencyId,
        outputCurrencyId: inputCurrencyId,
        inputCurrencyAmount: FractionUtils.serializeFractionToJSON(outputCurrencyAmount),
        outputCurrencyAmount: FractionUtils.serializeFractionToJSON(inputCurrencyAmount),
        orderKind: orderKind === OrderKind.SELL ? OrderKind.BUY : OrderKind.SELL,
      })
```

And we can't use the same code for TWAP orders as we use for Limit orders.

Comment (shoom3301) [reply src/modules/advancedOrders/hooks/useAdvancedOrdersActions.ts:61]: Good point. Anyway, we can unify it:
https://github.com/cowprotocol/cowswap/pull/2575

Comment (nenadV91) [reply src/modules/advancedOrders/hooks/useAdvancedOrdersActions.ts]: Meged

**Review by shoom3301 — APPROVED** [review]

**Comment by elena-zh** [comment]
Switch token button does not work when
1. One of tokens is not set
2. for wrap-unwrap flow

**Pull request merged**